File: tundra-cli/benchmarks/summing/sum.cs
// File: sum_squares/sum_squares.cs
// Compute sum of squares from 1..N (N = 2_000_000)

using System;

class SumSquares {
    static long SumSquaresMethod(int n) {
        long total = 0L;
        int i = 1;
        while (i <= n) {
            long ii = (long)i * (long)i;
            total += ii;
            i++;
        }
        return total;
    }

    static void Main(string[] args) {
        int n = 3000000;
        long result = SumSquaresMethod(n);
        Console.WriteLine(result);
    }
}

File: tundra-cli/benchmarks/summing/sum.rb
# File: sum_squares/sum_squares.rb
# Compute sum of squares from 1..N (N = 2_000_000)

def sum_squares(n)
  total = 0
  i = 1
  while i <= n
    ii = i * i
    total += ii
    i += 1
  end
  total
end

def main
  n = 3_000_000
  result = sum_squares(n)
  puts result
end

main
